DigitalLearningSolutions.Web/ViewModels/TrackingSystem/Centre/Reports/EditFiltersViewModel.cs

```diff
@@ -0,0 +1,167 @@
+            ActivityFilterData filterData,
+            int userCategoryId,
+            IEnumerable<(int, string)> jobGroupOptions,
+            IEnumerable<(int, string)> courseCategoryOptions,
+            IEnumerable<(int, string)> courseOptions,
+            DateTime dataStartDate
+        )
+        {
+            JobGroupId = filterData.JobGroupId;
+
+            if (filterData.CustomisationId.HasValue)
+            {
+                FilterType = CourseFilterType.Course;
+            }
+            else
+            {
+                FilterType = filterData.CourseCategoryId.HasValue
+                    ? CourseFilterType.CourseCategory
+                    : CourseFilterType.None;
+            }
+
+            CourseCategoryId = filterData.CourseCategoryId;
+            CustomisationId = filterData.CustomisationId;
+            StartDay = filterData.StartDate.Day;
+            StartMonth = filterData.StartDate.Month;
+            StartYear = filterData.StartDate.Year;
+            EndDay = filterData.EndDate?.Day;
+            EndMonth = filterData.EndDate?.Month;
+            EndYear = filterData.EndDate?.Year;
+            EndDate = filterData.EndDate.HasValue;
+            ReportInterval = filterData.ReportInterval;
+            DataStart = dataStartDate;
+
+            SetUpDropdownOptions(jobGroupOptions, courseCategoryOptions, courseOptions, userCategoryId);
+        }
+
+        public int? JobGroupId { get; set; }
+        public CourseFilterType FilterType { get; set; }
+
+        public int? CourseCategoryId
+        {
+            get => FilterType == CourseFilterType.CourseCategory ? courseCategoryId : null;
+            set => courseCategoryId = value;
+        }
+
+        public int? CustomisationId
+        {
+            get => FilterType == CourseFilterType.Course ? customisationId : null;
+            set => customisationId = value;
+        }
+
+        private int? customisationId { get; set; }
+        public int? StartDay { get; set; }
+        public int? StartMonth { get; set; }
+        public int? StartYear { get; set; }
+        public int? EndDay { get; set; }
+        public int? EndMonth { get; set; }
+        public int? EndYear { get; set; }
+        public bool EndDate { get; set; }
+        public ReportInterval ReportInterval { get; set; }
+        public DateTime DataStart { get; set; }
+        public bool CanFilterCourseCategories { get; set; }
+
+        public IEnumerable<SelectListItem> JobGroupOptions { get; set; } = new List<SelectListItem>();
+        public IEnumerable<SelectListItem> CourseCategoryOptions { get; set; } = new List<SelectListItem>();
+        public IEnumerable<SelectListItem> CustomisationOptions { get; set; } = new List<SelectListItem>();
+        public IEnumerable<SelectListItem> ReportIntervalOptions { get; set; } = new List<SelectListItem>();
+
+        public IEnumerable<ValidationResult> Validate(ValidationContext validationContext)

```

Comment:
This is very long - could we split this up a bit? Also, could some of this live in the DateValidator?